koku/api/query_params.py

```diff
@@ -157,60 +160,19 @@ def __str__(self):
-        if value.startswith(AND_TAG_PREFIX):
-            return value[len(AND_TAG_PREFIX) :]  # noqa: E203
-        if value.startswith(OR_TAG_PREFIX):
-            return value[len(OR_TAG_PREFIX) :]  # noqa: E203
-
-    def _process_tag_query_params(self, query_params):
-        """Reduce the set of tag keys based on those being queried."""
-        param_tag_keys = set()
-        for key, value in query_params.items():
-            if not isinstance(value, (dict, list)):
-                value = [value]
-            for inner_key in value:
-                stripped_key = self._strip_tag_prefix(inner_key)
-                if stripped_key in self.tag_keys:
-                    param_tag_keys.add(inner_key)
-            stripped_key = self._strip_tag_prefix(key)
-            if stripped_key in self.tag_keys:
-                param_tag_keys.add(key)
-        return param_tag_keys
-
-    # TODO: This function will be needed for below
-    # def _strip_aws_category_prefix(self, value):
-    #     if AWS_CATEGORY_PREFIX.replace(":", "") not in value:
-    #         return value
-    #     if value.startswith(AWS_CATEGORY_PREFIX):
-    #         return value[len(AWS_CATEGORY_PREFIX) :]  # noqa: E203
-    #     if value.startswith(AND_AWS_CATEGORY_PREFIX):
-    #         return value[len(AND_AWS_CATEGORY_PREFIX) :]  # noqa: E203
-    #     if value.startswith(OR_AWS_CATEGORY_PREFIX):
-    #         return value[len(OR_AWS_CATEGORY_PREFIX) :]  # noqa: E203
-
-    def _process_aws_category_query_params(self, query_params):
-        """Reduce the set of aws categories based on those being queried."""
-        param_aws_category_keys = set()
-        for key, value in query_params.items():
-            if not isinstance(value, (dict, list)):
-                value = [value]
-            for inner_key in value:
-                if AWS_CATEGORY_PREFIX in inner_key:
-                    param_aws_category_keys.add(inner_key)
-                # TODO: Remove when we have better serializer logic
-                # for aws_category keys
-                # stripped_key = self._strip_aws_category_prefix(inner_key)
-                # if stripped_key in self.aws_category_keys:
-            # stripped_key = self._strip_aws_category_prefix(key)
-            # if stripped_key in self.aws_category_keys:
-            #     param_aws_category_keys.add(key)
-        return param_aws_category_keys
+    def _strip_prefix(self, key, common_substring, prefix_list):
+        """Strip the prefixes from the key.
+
+        Args:
+            key: key passed in the query param
+            common_substring: found in all elements of prefix list of strings
+            prefix_list: possible prefixes to be found in query params
+        """
+        if common_substring not in key:
+            return key
+        for prefix in prefix_list:
+            if key.startswith(prefix):
+                return key[len(prefix) :]  # noqa: E203
```

Comment:
```suggestion
                return key[len(prefix):] 
```

Reply:
So, this one of those times where black & flake8 are fighting each other. if I try to commit this it will change it then flake8 will complain:
```
Flake8..........................................................................Failed
- hook id: flake8
- exit code: 1

koku/api/query_params.py:175:39: E203 whitespace before ':'

Check pipenv's Pipfile.lock to be consistent with Pipfile...(no files to check)Skipped
Validate Grafana Dashboards.................................(no files to check)Skipped
(koku) cmyers@cmyers1-mac koku % git diff
diff --git a/koku/api/query_params.py b/koku/api/query_params.py
index 53589f2f..710d3298 100644
--- a/koku/api/query_params.py
+++ b/koku/api/query_params.py
@@ -172,7 +172,7 @@ class QueryParameters:
             return key
         for prefix in prefix_list:
             if key.startswith(prefix):
-                return key[len(prefix):]
+                return key[len(prefix) :]
```

Reply:
Well that's pretty frustrating. We should just disable E203 in our flake8 config.